src/k8s/pkg/utils/file_operation.go

```diff
@@ -0,0 +1,73 @@
+package utils
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+// FileOperations is a struct that helps in perfoming file operations like
+// backup and write multiple files.
+type FileOperations struct {
+	BackupPath  string
+	SourcePath  string
+	Content     []byte
+	Permissions os.FileMode
+}
+
+// BackupFiles backs up the files in the operations slice.
+func BackupFiles(operations []FileOperations) error {
+	for _, op := range operations {
+		if err := backupFile(op.SourcePath, op.BackupPath); err != nil {
+			return fmt.Errorf("backup failed: %w", err)
```

Comment:
it probably makes sense to include the SourcePath and BackupPath in the error message